tests/Main.hs

```diff
@@ -49,139 +49,5116 @@ dimensionsTest =
+            , "28.0"
+            , "29.0"
+            , "30.0"
+            , "31.0"
+            , "32.0"
+            , "33.0"
+            , "34.0"
+            , "35.0"
+            , "36.0"
+            , "37.0"
+            , "38.0"
+            , "39.0"
+            , "40.0"
+            , "41.0"
+            , "42.0"
+            , "43.0"
+            , "44.0"
+            , "45.0"
+            , "46.0"
+            , "47.0"
+            , "48.0"
+            , "49.0"
+            , "50.0"
+            , "3.14"
+            , "2.22"
+            , "8.55"
+            , "23.3"
+            , "12.22222235049451"
+            ]
+        beforeParse :: [T.Text]
+        beforeParse =
+            [ "To"
+            , "protect"
+            , "the"
+            , "world"
+            , "from"
+            , "devastation"
+            , "1.0"
+            , "2.0"
+            , "3.0"
+            , "4.0"
+            , "5.0"
+            , "6.0"
+            , "7.0"
+            , "8.0"
+            , "9.0"
+            , "10.0"
+            , "11.0"
+            , "12.0"
+            , "13.0"
+            , "14.0"
+            , "15.0"
+            , "16.0"
+            , "17.0"
+            , "18.0"
+            , "19.0"
+            , "20.0"
+            , "21.0"
+            , "22.0"
+            , "23.0"
+            , "24.0"
+            , "25.0"
+            , "26.0"
+            , "27.0"
+            , "28.0"
+            , "29.0"
+            , "30.0"
+            , "31.0"
+            , "32.0"
+            , "33.0"
+            , "34.0"
+            , "35.0"
+            , "36.0"
+            , "37.0"
+            , "38.0"
+            , "39.0"
+            , "40.0"
+            , "41.0"
+            , "42.0"
+            , "43.0"
+            , "44.0"
+            , "45.0"
+            , "46.0"
+            , "47.0"
+            , "48.0"
+            , "49.0"
+            , "50.0"
+            , "3.14"
+            , "2.22"
+            , "8.55"
+            , "23.3"
+            , "12.22222235049451"
+            ]
+        expected = DI.BoxedColumn $ V.fromList afterParse
+        actual = D.parseDefault 10 True "%Y-%m-%d" $ DI.fromVector $ V.fromList beforeParse
+     in TestCase
+            ( assertEqual
+                "Correctly parses a mixture of Texts and Doubles as BoxedColumn of Texts"
+                expected
+                actual
+            )
+
+parseDatesAndTextsAsTexts :: Test
+parseDatesAndTextsAsTexts =
+    let afterParse :: [T.Text]
+        afterParse =
+            [ "2020-02-12"
+            , "2020-02-13"
+            , "2020-02-14"
+            , "2020-02-15"
+            , "2020-02-16"
+            , "2020-02-17"
+            , "2020-02-18"
+            , "2020-02-19"
+            , "2020-02-20"
+            , "2020-02-21"
+            , "2020-02-22"
+            , "2020-02-23"
+            , "2020-02-24"
+            , "2020-02-25"
+            , "2020-02-26"
+            , "Jessie"
```

Comment:
Same here. I wanna say to the user...these are mostly dates minus some outliers. Especially for some random CSV that a user is exploring.